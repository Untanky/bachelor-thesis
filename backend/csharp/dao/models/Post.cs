using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dao.models {
  [Table("post")]
  public class Post
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string Title { get; set; }

    public string Description { get; set; }

    public Post() {

    }

    public Post(string title, string description) {
      this.Title = title;
      this.Description = description;
    }

    // override object.Equals
    public override bool Equals(object obj)
    {
        //
        // See the full list of guidelines at
        //   http://go.microsoft.com/fwlink/?LinkID=85237
        // and also the guidance for operator== at
        //   http://go.microsoft.com/fwlink/?LinkId=85238
        //
        
        if (obj == null || GetType() != obj.GetType())
        {
            return false;
        }

        var other = (Post)obj;
        
        return this.Id == other.Id 
          && this.Title.Equals(other.Title)
          && this.Description.Equals(other.Description);
    }
  }
}
