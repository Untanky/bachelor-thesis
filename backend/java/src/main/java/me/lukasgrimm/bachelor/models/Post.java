package me.lukasgrimm.bachelor.models;

import javax.persistence.*;

@Entity
@Table(name = "post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", unique = true)
    private long id;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    private String url;


    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return String.format("/post/%d", id);
    }
}
