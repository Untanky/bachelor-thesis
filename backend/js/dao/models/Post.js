import { DataTypes, Model } from 'sequelize';
import sequelize from '../config/sequelize';

class User extends Model {}

User.init({
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
  },
  description: {
    type: DataTypes.STRING,
  },
}, {
  sequelize,
});
