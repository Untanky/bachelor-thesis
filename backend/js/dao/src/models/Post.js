import { DataTypes, Model } from 'sequelize';
import sequelize from '../config/sequelize';

class Post extends Model {}

Post.init({
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
  },
  description: {
    type: DataTypes.STRING,
  },
}, {
  sequelize,
  tableName: 'post',
  timestamps: false,
});

export default Post;
