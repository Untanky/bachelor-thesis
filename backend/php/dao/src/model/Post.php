<?php

namespace model;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity @ORM\Table(name="post")
 */
class Post implements \JsonSerializable
{
    /** @ORM\ID @ORM\Column(type="integer") @ORM\GeneratedValue */
    private $id;

    /** @ORM\Column(type="string") */
    private $title;

    /** @ORM\Column(type="string") */
    private $description;

    public function __construct()
    {
    }

    public static function fromArray($data) {
        $post = new Post();
        if(property_exists($data, 'id')) {
            $post->id = $data->id;
        }
        if(property_exists($data, 'title')) {
            $post->title = $data->title;
        }
        if(property_exists($data, 'description')) {
            $post->description = $data->description;
        }
        return $post;
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @return mixed
     */
    public function getTitle()
    {
        return $this->title;
    }

    /**
     * @param mixed $title
     */
    public function setTitle($title)
    {
        $this->title = $title;
    }

    /**
     * @return mixed
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param mixed $description
     */
    public function setDescription($description)
    {
        $this->description = $description;
    }

    public function getUrl() {
        return "/api/blog/post/" . $this->getId();
    }

    public function jsonSerialize()
    {
        return
            [
                'id'   => $this->getId(),
                'title' => $this->getTitle(),
                'description' => $this->getDescription(),
                'url' => $this->getUrl()
            ];
    }
}