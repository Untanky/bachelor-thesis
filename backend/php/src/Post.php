<?php

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity @ORM\Table(name="post")
 */
class Post
{
    /** @ORM\ID @ORM\Column(type="integer") @ORM\GeneratedValue */
    private $id;

    /** @ORM\Column(type="string")  */
    private $title;

    /** @ORM\Column(type="string")  */
    private $description;

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @return mixed
     */
    public function getTitle()
    {
        return $this->title;
    }

    /**
     * @param mixed $title
     */
    public function setTitle($title)
    {
        $this->title = $title;
    }

    /**
     * @return mixed
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param mixed $description
     */
    public function setDescription($description)
    {
        $this->description = $description;
    }
}