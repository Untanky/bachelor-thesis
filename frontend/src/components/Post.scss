@import "../theme";

.post {
  text-align: left;
  padding: 2em;
  border: 2px solid $htw-green;
  border-radius: 2em;

  .post-title {
    margin: 0;
  }

  .button-container {
    > *:not(:first-child)  {
      margin-left: 0.5em;
    }

    .delete-link {
      font-size: 16px;
    }
  }
}