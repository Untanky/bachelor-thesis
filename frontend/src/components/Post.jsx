import React from 'react';

import ButtonLink from './ButtonLink';

import './Post.scss';

const Post = ( { post } ) => {
  return (
    <div className="post">
      <h2 className="post-title">{post.title}</h2>
      <p className="post-description">{post.description}</p>
      <ButtonLink
        className="edit-link"
        to={`/post/edit/${post.id}`}
        text="Edit post"
      />
    </div>
  )
};

export default Post;
