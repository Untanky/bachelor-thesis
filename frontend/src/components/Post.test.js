import React from 'react';
import Post from './Post';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

const post = {
  id: 1,
  title: 'Test title',
  description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
  numberOfComments: 6,
  url: 'post/1',
  deleteClicked: () => {},
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Post post={post} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
