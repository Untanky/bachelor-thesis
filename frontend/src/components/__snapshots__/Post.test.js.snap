// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="post"
>
  <h2
    className="post-title"
  >
    Test title
  </h2>
  <p
    className="post-description"
  >
    This is the description for the test post. It contains vital data for the post and puts it into context.
  </p>
  <div
    className="button-container"
  >
    <a
      className="button edit-link"
      href="/post/edit/1"
      onClick={[Function]}
    >
      Edit Post
    </a>
    <button
      className="button delete delete-link"
      onClick={[Function]}
    >
      Delete post
    </button>
  </div>
</div>
`;
